debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbEventMetadataProvider.java

```diff
@@ -32,6 +33,10 @@ public Instant getEventTimestamp(DataCollectionId source, OffsetContext offset,
         if (source == null) {
             return null;
         }
+        if (sourceInfo.schema().field(SourceInfo.TIMESTAMP) != null) {
+            final Integer timestamp = sourceInfo.getInt32(SourceInfo.TIMESTAMP);
+            return timestamp == null ? null : Conversions.toInstantFromMicros(timestamp);
```

Comment:
Isn't the units for the "sec" field second, not microseconds?

Reply:
You're absolutely correct, the time value in the source-info block is seconds since epoch and so this should be changed.